internal/cmd/resources/resources.go

```diff
@@ -0,0 +1,174 @@
+// Copyright Mia srl
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/mia-platform/miactl/internal/client"
+	"github.com/mia-platform/miactl/internal/clioptions"
+	"github.com/mia-platform/miactl/internal/resources"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+const (
+	CronJobResourceType  = "cronjob"
+	CronJobsResourceType = "cronjobs"
+
+	DeploymentResourceType  = "deployment"
+	DeploymentsResourceType = "deployments"
+
+	JobResourceType  = "job"
+	JobsResourceType = "jobs"
+
+	PodResourceType  = "pod"
+	PodsResourceType = "pods"
+
+	ServiceResourceType  = "service"
+	ServicesResourceType = "services"
+
+	listEndpointTemplate = "/api/projects/%s/environments/%s/%s/describe/"
+)
+
+var resourcesAvailable = []string{
+	CronJobResourceType,
+	CronJobsResourceType,
+	DeploymentResourceType,
+	DeploymentsResourceType,
+	JobResourceType,
+	JobsResourceType,
+	PodResourceType,
+	PodsResourceType,
+	ServiceResourceType,
+	ServicesResourceType,
+}
+
+func ListCommand(o *clioptions.CLIOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list RESOURCE-TYPE ENVIRONMENT",
+		Short: "List Mia-Platform Console runtime resources",
+		Long: `List Mia-Platform Console runtime resources.
+
+A project on Mia-Platform Console once deployed can have one or more resource of different kinds associcated with one
+or more of its environments.`,
+		Args: cobra.ExactArgs(2),
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			return resourcesCompletions(args, toComplete), cobra.ShellCompDirectiveNoFileComp
```

Comment:
With this, there is the autocomplete with tab using the cli?

Reply:
Yep, it will autocomplete only the service type for now